corehq/apps/hqwebapp/two_factor_gateways.py

```diff
@@ -108,6 +111,25 @@ def get_ip_address():
         else:
             return None
 
+    def _check_for_exceeded_rate_limits(ip, num, user):
+        ip_window = two_factor_rate_limiter_per_ip.rate_limit_exceeded_for_window('ip:{}'.format(ip))
```

Comment:
these returned windows are just the rate_counter.key values, which I'm honestly still unsure how those are populated. I see this comment implies their values: 
https://github.com/dimagi/commcare-hq/blob/master/corehq/project_limits/rate_limiter.py#L59-L64

Reply:
Nice I see. Yeah I this this does work, and I think they're already in the correct order. They're tested in the order defined here: https://github.com/dimagi/commcare-hq/blob/4fcbc662a38090b4563f21acf51c6a2faeb534ec/corehq/project_limits/rate_limiter.py#L194-L201, and the strings 'week', 'day', etc are defined in `*_rate_counter` here: https://github.com/dimagi/commcare-hq/blob/cc93bc6ab718bde08e8d16e374d5cc25eff94958/corehq/project_limits/rate_counter/presets.py#L6-L39.